### PR TITLE
Announce 4.143

### PR DESCRIPTION
**Pull request by fredemmott**
Still waiting for MacOS builds for the other releases; needed a few
extra cherrypicks to build old versions.

### CODE REVIEW

File: _posts/2022-01-06-hhvm-4.143.markdown [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: "HHVM 4.143"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.143 is released! This release marks the end of support of 4.135;
+4.136&ndash;4.140 remain supported, as do th 4.102 and 4.128 LTS releases.
```
Comment (lexidor) [inline _posts/2022-01-06-hhvm-4.143.markdown:9]:
```suggestion
4.136&ndash;4.140 remain supported, as do the 4.102 and 4.128 LTS releases.
```

File: _posts/2022-01-06-hhvm-4.143.markdown [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+- If an interface or class is `<<__ConsistentConstruct>>` but does not define
+  a constructor, all subclass constructors must now be pure (`[]`).
+  Alternatively, define a constructor on the interface or class to specify
+  different capabilities. In previous versions, this can be enabled with
+  the `pure_default_consistent_constructors` flag in `.hhconfig`.
+- `DateTimeInterface::getTimestamp()` is now pure; any custom implementations
+  will need to also be pure.
+
+# Future Changes
```
Comment (lexidor) [inline _posts/2022-01-06-hhvm-4.143.markdown:31]: This is an empty heading.

File: _posts/2022-01-06-hhvm-4.143.markdown [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+4.136&ndash;4.140 remain supported, as do th 4.102 and 4.128 LTS releases.
+
+Additional patch releases have been made for 4.102, 4.128, and 4.135-4.140,
+which change the typechecker to ignore any files in `vendor/bin/`; this is
+required for compatibility with Composer 2.2.
+
+# Highlights
+
+- Compatible with Composer 2.2.
+- Added documentation on hover for `readonly` and `extends` keywords.
```
Comment (lexidor) [inline _posts/2022-01-06-hhvm-4.143.markdown:18]: I believe that the commit that adds hover info for `async` and friends is also in 4.143. The commit that added this was made two commits after the vendor/bin commit.
```suggestion
- Added documentation on hover for `async` `async {...}`, `await`, `concurrent`, `extends`, and `readonly`.
```

https://github.com/facebook/hhvm/commit/e2be871af1b3ea6ac252909380569fa5d0593338

**Comment by fredemmott** [comment]
Thanks - still waiting for mac builds :(

File: _posts/2022-01-06-hhvm-4.143.markdown [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+title: "HHVM 4.143"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.143 is released! This release marks the end of support of 4.135;
+4.136&ndash;4.140 remain supported, as do the 4.102 and 4.128 LTS releases.
```
Comment (Atry) [inline _posts/2022-01-06-hhvm-4.143.markdown:9]: Shall we mention what happened to 4.141 and 4.142?

**Review by Atry — APPROVED** [review]

**Pull request merged**